app/celery/tasks.py

```diff
@@ -226,6 +205,15 @@ def create_pdf_for_templated_letter(self: Task, encoded_letter_data):
     )
 
 
+def _prepare_pdf(letter_details, self):
+    def create_pdf_for_letter(letter_details, language, include_tag) -> BytesIO:
```

Comment:
Now I remembered - didn't we want to change this nested function and the same in `preview.py` into lambdas?

Reply:
no, ruff will fail checks against lambdas assigned to variables, saying nested functions are preferred

we'd previously started with lambdas, then switched to get tests to pass, so i squashed that out.

looking elsewhere, there's some discussion saying "If you need to assign the lambda to a name, use a def instead"

https://stackoverflow.com/a/134638


i guess we could inline it, for example:

```
def _prepare_pdf(letter_details, self):
    purpose = PDFPurpose.PRINT

    return generate_templated_pdf(
        letter_details,
        (
            lambda letter_details, language, include_tag: _create_pdf_for_letter(
                self, letter_details, language=language, include_notify_tag=include_tag
            )
        ),
        purpose,
    )
```

but i don't particularly love that. open to hear other thoughts tho!

Reply:
Ok, no worries then ☺️